File: 367-valid-perfect-square/valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        # Using Binary Search

        # low = 1
        # right = num

        # while low <= right:
        #     mid = (low+right)//2
        #     res = mid * mid
        #     if res == num:
        #         return True
        #     elif res < num:
        #         low = mid+1
        #     else:
        #         right = mid-1
        # return False

        a, b = math.modf(num ** 0.5)
        return a == 0

File: 367-valid-perfect-square/valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        long low = 1, right = num;
        while (low <= right){
            long mid = (low + right) / 2;
            long res = mid * mid;
            if (res == num) return true;
            else if (res < num) low = mid + 1;
            else right = mid - 1;
        }
        return false;
    }
}